### PR TITLE
This PR adds the updates in UI and of integration of Mess Section in the app: Feedback, Complaint, MOM, Tenders and their API integration

### PR DESCRIPTION
**Pull request by mtabishk**
Updated the UI of different pages in the Mess Section.
Integrated the backend APIs

Mess Feedback - UI and API Integration
Mess Complaint - UI and API Integration
Mess MOM - UI and API Integration
Mess Tender - UI and API Integration
Mess Menu - UI only

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1 +1,6 @@
-DJANGO_SECRET_KEY=secret key
\ No newline at end of file
+DJANGO_SECRET_KEY=secret key
```
Comment (ashutoshc8101) [inline .env.example:1]: Revert changes to this file.

File: api/mess/models.py [hunk 2]
```diff
@@ -2,71 +2,80 @@
...
+                             on_delete=models.CASCADE)
+    body = models.TextField(editable=True, null=False, blank=True)
+    mess_meal = models.CharField(max_length=50)
+    mess_type = models.ForeignKey(Mess, on_delete=models.CASCADE, null=True)
+    created_at = models.DateTimeField(auto_now=True)
+    status = models.BooleanField(default=False)
 
 # data base model for mess mom
+
+
```
Comment (ashutoshc8101) [inline api/mess/models.py:63]: Remove these blank lines.

File: api/mess/models.py [hunk 3]
```diff
@@ -2,71 +2,80 @@
...
+    date = models.DateField(default=None, null=True)
+    file = models.FileField(upload_to='mom/')
+    title = models.CharField(max_length=100)
     description = models.TextField()
-    created_at = models.DateTimeField(auto_now = True)
+    created_at = models.DateTimeField(auto_now=True)
 
 # data base model for mess tender
+
+
```
Comment (ashutoshc8101) [inline api/mess/models.py:73]: ditto

File: api/mess/views.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+from ast import Is
```
Comment (ashutoshc8101) [inline api/mess/views.py:1]: This import is not used. Make sure to not have unused imports.

File: api/mess/views.py [hunk 5]
```diff
@@ -7,7 +8,7 @@
 
 
 class MessMenu(APIView):
-    permission_classes = [IsAuthenticated]
+    #permission_classes = [IsAuthenticated]
```
Comment (ashutoshc8101) [inline api/mess/views.py:11]: Uncomment this line.

File: app/.env.symlink [hunk 6]
```diff
@@ -1 +1,6 @@
-../.env
\ No newline at end of file
+
```
Comment (ashutoshc8101) [inline app/.env.symlink:1]: Revert changes to this file.

File: app/.gitignore [hunk 7]
```diff
@@ -43,4 +43,7 @@ app.*.map.json
 # Android Studio will place build artifacts here
 /android/app/debug
 /android/app/profile
-/android/app/release
\ No newline at end of file
+/android/app/release
```
Comment (ashutoshc8101) [inline app/.gitignore:46]: Maybe this change is required for your development environment but should not be pushed to Github as it will break other people's oauth setup.

File: app/android/app/build.gradle [hunk 8]
```diff
@@ -26,7 +26,7 @@ apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion flutter.compileSdkVersion
+    compileSdkVersion 33
```
Comment (ashutoshc8101) [inline app/android/app/build.gradle:29]: Why this is required?

File: app/android/app/build.gradle [hunk 9]
```diff
@@ -52,10 +52,10 @@ android {
...
     signingConfigs {
         debug {
-            storeFile file('key.jks')
-            storePassword 'instituteapp'
-            keyAlias 'androiddebugkey'
-            keyPassword 'instituteapp'
+            storeFile file('upload-keystore.jks')
+            storePassword 'password'
+            keyAlias 'uploadkeystore'
+            keyPassword 'password'
```
Comment (ashutoshc8101) [inline app/android/app/build.gradle:58]: Revert this change.

File: app/lib/screens/mess/manager/complaints_mess_manager.dart [hunk 10]
```diff
@@ -1,30 +1,77 @@
...
+      },
+    );
+
+    if (response.statusCode == 200) {
+      final data = jsonDecode(response.body);
+      return data
+          .map<FeedbackModel>((json) => FeedbackModel.fromJson(json))
+          .toList();
+    } else {
+      throw Exception('Failed to load post');
```
Comment (ashutoshc8101) [inline app/lib/screens/mess/manager/complaints_mess_manager.dart:40]: Instead show an error to user that there is a problem with network connection. This can be done later.

File: app/lib/screens/mess/manager/complaints_mess_manager.dart [hunk 11]
```diff
@@ -1,30 +1,77 @@
...
+    } else {
+      throw Exception('Failed to get');
+    }
+  }
+
   Widget build(BuildContext context) {
     final size = MediaQuery.of(context).size;
     return Scaffold(
-      backgroundColor: Colors.blue,
+      backgroundColor: Color(0xFF00ABE9),
```
Comment (ashutoshc8101) [inline app/lib/screens/mess/manager/complaints_mess_manager.dart:66]: Let's stick with `Colors.blue` for consistent design.

File: app/lib/screens/mess/manager/feedback_mess_manager.dart [hunk 12]
```diff
@@ -1,26 +1,55 @@
...
       appBar: AppBar(
-        title: const Text(
+        title: Text(
           "Feedbacks",
           style: TextStyle(
             fontSize: 20,
           ),
         ),
-        backgroundColor: Colors.blue,
+        backgroundColor: Color(0xFF00ABE9),
```
Comment (ashutoshc8101) [inline app/lib/screens/mess/manager/feedback_mess_manager.dart:52]: ditto

File: app/lib/screens/mess/student/feedback_mess_student.dart [hunk 13]
```diff
@@ -52,22 +53,23 @@ class _FeedbackMessState extends State<FeedbackMess> {
     if (response.statusCode == 200) {
       return Future.value(true);
     }
+
```
Comment (ashutoshc8101) [inline app/lib/screens/mess/student/feedback_mess_student.dart:56]: Remove this line.

File: app/lib/screens/mess/student/mess_student.dart [hunk 14]
```diff
@@ -168,3 +166,102 @@ class _MessStudentScreenState extends State<MessStudentScreen> {
     );
   }
 }
+    // return FutureBuilder(
+    //     future: fetchMessData(),
+    //     builder: (BuildContext context,
+    //         AsyncSnapshot<Map<String, List<Mess>>> snapshot) {
```
Comment (ashutoshc8101) [inline app/lib/screens/mess/student/mess_student.dart:172]: Make sure to remove residual code before asking for review

File: app/lib/services/auth.service.dart [hunk 15]
```diff
@@ -25,6 +25,7 @@ class AuthService with ChangeNotifier {
 
   AuthService() {
     _googleSignIn.onCurrentUserChanged.listen((GoogleSignInAccount? account) {
+      print(account);
```
Comment (ashutoshc8101) [inline app/lib/services/auth.service.dart:28]: Remove residual print statements.

**Review by ashutoshc8101 — CHANGES_REQUESTED** [review]
Left few comments. Flutter analyze is failing, look into that. Also add screenshots and videos of newly added features.

File: app/android/app/build.gradle (hunk 8 above)
Comment (mtabishk) [reply app/android/app/build.gradle:29]: This is required for a package flutter_downloader

File: .env.example (hunk 1 above)
Comment (mtabishk) [reply .env.example]: reverted

File: api/mess/models.py (hunk 2 above)
Comment (mtabishk) [reply api/mess/models.py]: removed

File: api/mess/views.py (hunk 4 above)
Comment (mtabishk) [reply api/mess/views.py]: removed

File: api/mess/views.py (hunk 5 above)
Comment (mtabishk) [reply api/mess/views.py]: done

File: app/lib/screens/mess/student/feedback_mess_student.dart (hunk 13 above)
Comment (mtabishk) [reply app/lib/screens/mess/student/feedback_mess_student.dart]: removed

File: app/lib/screens/mess/student/mess_student.dart (hunk 14 above)
Comment (mtabishk) [reply app/lib/screens/mess/student/mess_student.dart]: removed

File: app/lib/services/auth.service.dart (hunk 15 above)
Comment (mtabishk) [reply app/lib/services/auth.service.dart]: removed

File: app/.env.symlink (hunk 6 above)
Comment (iraapp) [reply app/.env.symlink:1]: Still Not done

File: app/.gitignore [hunk 16]
```diff
@@ -43,4 +43,6 @@ app.*.map.json
 # Android Studio will place build artifacts here
 /android/app/debug
 /android/app/profile
-/android/app/release
\ No newline at end of file
+
+# keystore related
```
Comment (iraapp) [inline app/.gitignore:47]: Not reverted.

**Review by iraapp — CHANGES_REQUESTED** [review]

File: app/.gitignore (hunk 16 above)
Comment (mtabishk) [reply app/.gitignore:47]: ok

**Comment by ashutoshc8101** [comment]
@mtabishk All review comments were not addreseed. Please take a look again.

**Comment by mtabishk** [comment]
@ashutoshc8101 Made the review changes

**Comment by ashutoshc8101** [comment]
@mtabishk `upload-keystore.jks` shouldn't be uploaded. Please remove it from the PR.

**Review by ashutoshc8101 — APPROVED** [review]
